integration-tests/deployment/devenv/don.go

```diff
@@ -0,0 +1,302 @@
+	var err error
+	for sel, chain := range chains {
+		for _, node := range don.Nodes {
+			// if node is bootstrap, no need to fund it
+			if node.multiAddr != "" {
+				continue
+			}
+			accountAddr, ok := node.AccountAddr[sel]
+			if !ok {
+				err = multierror.Append(err, fmt.Errorf("node %s has no account address for chain %d", node.Name, sel))
+				continue
+			}
+			if err1 := FundAddress(ctx, chain.DeployerKey, common.HexToAddress(accountAddr), amount, chain); err1 != nil {
+				err = multierror.Append(err, err1)
+			}
+		}
+	}
+	return err
+}
+
+func (don *DON) CreateSupportedChains(ctx context.Context, chains []ChainConfig) error {
+	var err error
+	for i, node := range don.Nodes {
+		if err1 := node.CreateCCIPOCR2SupportedChains(ctx, chains); err1 != nil {
+			err = multierror.Append(err, err1)
+		}
+		don.Nodes[i] = node
+	}
+	return err
+}
+
+func NewRegisteredDON(ctx context.Context, nodeInfo []NodeInfo, jd JobDistributor) (*DON, error) {
+	don := &DON{
+		Nodes: make([]Node, 0),
+	}
+	for i, info := range nodeInfo {
+		if info.Name == "" {
+			info.Name = fmt.Sprintf("node-%d", i)
+		}
+		node, err := NewNode(info)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create node %d: %w", i, err)
+		}
+		// node Labels so that it's easier to query them
+		if info.IsBootstrap {
+			// create multi address for OCR2, applicable only for bootstrap nodes
+
+			node.multiAddr = fmt.Sprintf("%s:%s", info.CLConfig.InternalIP, info.P2PPort)
+			// no need to set admin address for bootstrap nodes, as there will be no payment
+			node.adminAddr = ""
+			node.labels = append(node.labels, &ptypes.Label{
+				Key:   NodeLabelType,
+				Value: pointer.ToString(NodeLabelBootstrap),
+			})
+		} else {
+			// multi address is not applicable for non-bootstrap nodes
+			// explicitly set it to empty string to denote that
+			node.multiAddr = ""
+			node.labels = append(node.labels, &ptypes.Label{
+				Key:   NodeLabelType,
+				Value: pointer.ToString(NodeLabelPlugin),
+			})
+		}
+		// Set up Job distributor in node and register node with the job distributor
+		err = node.SetUpAndLinkJobDistributor(ctx, jd)
+		if err != nil {
+			return nil, fmt.Errorf("failed to set up job distributor in node %s: %w", info.Name, err)
+		}
+
+		don.Nodes = append(don.Nodes, *node)
+	}
+	return don, nil
+}
+
+func NewNode(nodeInfo NodeInfo) (*Node, error) {
+	gqlClient, err := client.New(nodeInfo.CLConfig.URL, client.Credentials{
+		Email:    nodeInfo.CLConfig.Email,
+		Password: nodeInfo.CLConfig.Password,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create FMS client: %w", err)
+	}
+	return &Node{
+		gqlClient: gqlClient,
+		Name:      nodeInfo.Name,
+		adminAddr: nodeInfo.AdminAddr,
+	}, nil
+}
+
+type Node struct {
+	NodeId      string
+	JDId        string
+	Name        string
+	AccountAddr map[uint64]string // chain selector to account address
+	gqlClient   client.Client
+	labels      []*ptypes.Label
+	adminAddr   string
+	multiAddr   string
```

Comment:
This needs to be documented, I don't recall what this is and the name isn't really helpful at all in helping me remember it.